src/lava/proc/conv/utils.py

```diff
@@ -0,0 +1,236 @@
+        ) / stride[0] + 1
+    ).astype(int)
+    y_out = np.floor(
+        (
+            input_shape[1] + 2 * padding[1]
+            - dilation[1] * (kernel_size[1] - 1) - 1
+        ) / stride[1] + 1
+    ).astype(int)
+
+    if x_out < 1:
+        print(f'{input_shape=}')
+        print(f'{out_channels=}')
+        print(f'{kernel_size=}')
+        print(f'{stride=}')
+        print(f'{padding=}')
+        print(f'{dilation=}')
+        print(x_out, y_out, out_channels)
+        raise Exception(
+            f'Found output x dimension (={x_out}) to be less than 1.'
+            f'Check your convolution sizes.'
+        )
+    if y_out < 1:
+        print(f'{input_shape=}')
+        print(f'{out_channels=}')
+        print(f'{kernel_size=}')
+        print(f'{stride=}')
+        print(f'{padding=}')
+        print(f'{dilation=}')
+        print(x_out, y_out, out_channels)
+        raise Exception(
+            f'Found output y dimension (={y_out}) to be less than 1.'
+            f'Check your convolution sizes.'
+        )
+
+    return x_out, y_out, out_channels
+
+
+# This might be needed later. The reverse mapping is ambiguous though
+# def input_shape(
+#     output_shape, in_channels, kernel_size, stride, padding, dilation
+# ):
+#     pass
+
+
+def conv(input, weight, kernel_size, stride, padding, dilation, groups):
+    """Convolution implementation
+
+    Parameters
+    ----------
+    input : 3 dimensional np array
+        convolution input.
+    weight : 4 dimensional np array
+        convolution kernel weight.
+    kernel_size : 2 element tuple, list, or array
+        convolution kernel size in XY/WH format.
+    stride : 2 element tuple, list, or array
+        convolution stride in XY/WH format.
+    padding : 2 element tuple, list, or array
+        convolution padding in XY/WH format.
+    dilation : 2 element tuple, list, or array
+        dilation of convolution kernel in XY/WH format.
+    groups : int
+        number of convolution groups.
+
+    Returns
+    -------
+    3 dimensional np array
+        convolution output
+    """
+    if TORCH_IS_AVAILABLE:
+        # with torch.no_grad():  # this seems to cause problems
+        output = F.conv2d(
+            torch.unsqueeze(  # torch expects a batch dimension NCHW
+                torch.FloatTensor(input.transpose([2, 1, 0])),
+                dim=0,
+            ),
+            torch.FloatTensor(
+                # torch acutally does correlation
+                # so flipping the spatial dimension of weight
+                # copy() is needed because
+                # torch cannot handle negative stride
+                weight[:, ::-1, ::-1].transpose([0, 3, 2, 1]).copy()
+            ),
+            stride=stride[::-1].tolist(),
+            padding=padding[::-1].tolist(),
+            dilation=dilation[::-1].tolist(),
+            groups=groups
+        )[0].cpu().data.numpy().transpose([2, 1, 0])
+    else:
+        output = conv_scipy(
```

Comment:
Would numpy.convolve also work? That way we might be able to reduce mandatory requirements for /lava

Reply:
numpy.convolve only works for 1d convolution. Besides we would need scipy for sparse matrix representation at some point.

Reply:
np.convolution only works for 1d data. We would need to use scipy for sparse matrix at some point anyway.